tests/__tests__/lib/tools/BaseDerivedCtorRegistry.js

```diff
@@ -0,0 +1,1211 @@
+                ['Delta', class Delta {}],
+              ]),
+            }),
+            classNameGenerator: DerivedClassNameGenerator.create({
+              className: 'BaseAppBeta',
+              fixedPrefix: 'BaseApp',
+            }),
+          },
+        },
+      ]
+
+      describe('#SuperCtor', () => {
+        test.each(cases)('Ctor: $input.SuperCtor.name', ({ input }) => {
+          const registry = new BaseDerivedCtorRegistry(input)
+
+          expect(registry)
+            .toHaveProperty('SuperCtor', input.SuperCtor)
+        })
+      })
+
+      describe('#pool', () => {
+        test.each(cases)('pool: $input.pool', ({ input }) => {
+          const registry = new BaseDerivedCtorRegistry(input)
+
+          expect(registry)
+            .toHaveProperty('pool', input.pool)
+        })
+      })
+
+      describe('#classNameGenerator', () => {
+        test.each(cases)('classNameGenerator: $input.classNameGenerator', ({ input }) => {
+          const registry = new BaseDerivedCtorRegistry(input)
+
+          expect(registry)
+            .toHaveProperty('classNameGenerator', input.classNameGenerator)
+        })
+      })
+    })
+  })
+})
+
+describe('BaseDerivedCtorRegistry', () => {
+  describe('.create()', () => {
+    describe('to be instance of own class', () => {
+      describe('with full parameters', () => {
+        const cases = [
+          {
+            input: {
+              SuperCtor: class BaseAlpha {},
+              pool: DynamicDerivedCtorPool.create(),
+              fixedPrefix: 'Base',
+            },
+          },
+          {
+            input: {
+              SuperCtor: class BaseAppBeta {},
+              pool: DynamicDerivedCtorPool.create(),
+              fixedPrefix: 'BaseApp',
+            },
+          },
+        ]
+
+        test.each(cases)('fixedPrefix: $input.fixedPrefix', ({ input }) => {
+          const registry = BaseDerivedCtorRegistry.create(input)
+
+          expect(registry)
+            .toBeInstanceOf(BaseDerivedCtorRegistry)
+        })
+      })
+
+      describe('with omitted parameters', () => {
+        const cases = [
+          {
+            input: {
+              SuperCtor: class BaseAlpha {},
+              fixedPrefix: 'Base',
+            },
+          },
+          {
+            input: {
+              SuperCtor: class BaseAppBeta {},
+              fixedPrefix: 'BaseApp',
+            },
+          },
+        ]
+
+        test.each(cases)('fixedPrefix: $input.fixedPrefix', ({ input }) => {
+          const registry = BaseDerivedCtorRegistry.create(input)
+
+          expect(registry)
+            .toBeInstanceOf(BaseDerivedCtorRegistry)
+        })
+      })
+    })
+
+    describe('to call constructor', () => {
+      const BaseAlpha = class {}
+      const BaseAppBeta = class {}
+      const BaseExtraGamma = class {}
+
+      const alphaCtorPool = DynamicDerivedCtorPool.create()
+      const BetaCtorPool = DynamicDerivedCtorPool.create()
```

Comment:
```suggestion
      const betaCtorPool = DynamicDerivedCtorPool.create()
```
細かい部分ですが、インスタンスなので camelCase でしょか